### PR TITLE
Added C# code (.NET Core) execution without tests

### PR DESCRIPTION
**Pull request by gogo4ds**
Implement simple execution of C# code (.NET Core) without tests
Closes https://github.com/SoftUni-Internal/interactive-issues/issues/56

### CODE REVIEW

File: OJS.Workers.ExecutionStrategies/DotNetCoreCompileExecuteAndCheckExecutionStrategy.cs [hunk 1]
```diff
@@ -35,27 +35,19 @@ public class DotNetCoreCompileExecuteAndCheckExecutionStrategy : ExecutionStrate
         {
             var result = new ExecutionResult<TestResult>();
 
-            // Compile the file
-            var compilerResult = this.ExecuteCompiling(executionContext, this.GetCompilerPathFunc, result);
-            if (!compilerResult.IsCompiledSuccessfully)
+            if (!this.ExecuteCompiling(executionContext, this.GetCompilerPathFunc, result, out var compilerResult))
```
Comment (jicata) [inline OJS.Workers.ExecutionStrategies/DotNetCoreCompileExecuteAndCheckExecutionStrategy.cs]: Extract into a variable for improved readability

File: OJS.Workers.ExecutionStrategies/DotNetCoreCompileExecuteAndCheckExecutionStrategy.cs [hunk 2]
```diff
@@ -78,6 +70,56 @@ public class DotNetCoreCompileExecuteAndCheckExecutionStrategy : ExecutionStrate
             return result;
         }
 
+        protected override IExecutionResult<RawResult> ExecuteNonCompetitive(IExecutionContext<string> executionContext)
```
Comment (jicata) [inline OJS.Workers.ExecutionStrategies/DotNetCoreCompileExecuteAndCheckExecutionStrategy.cs]: I'm not sure if this is the correct place, but naming can be improved here. I can't really tell what `RawResult` is supposed to be. Maybe something like `SimpleResult` would be an improvement?

 Also `ExecuteNonCompetitive` suggests that there is something that defines execution as competitive/non-competitive and I wonder what could that be?

File: OJS.Workers.ExecutionStrategies/ExecutionStrategy.cs [hunk 3]
```diff
@@ -92,8 +92,7 @@ protected ExecutionStrategy(int baseTimeUsed, int baseMemoryUsed)
             var result = new ExecutionResult<TestResult>();
 
             // Compile the file
-            var compilerResult = this.ExecuteCompiling(executionContext, getCompilerPathFunc, result);
-            if (!compilerResult.IsCompiledSuccessfully)
+            if (!this.ExecuteCompiling(executionContext, getCompilerPathFunc, result, out var compilerResult))
```
Comment (jicata) [inline OJS.Workers.ExecutionStrategies/ExecutionStrategy.cs]: Extract into a variable for improved readability

File: OJS.Workers.ExecutionStrategies/DotNetCoreCompileExecuteAndCheckExecutionStrategy.cs (hunk 2 above)
Comment (gogo4ds) [reply OJS.Workers.ExecutionStrategies/DotNetCoreCompileExecuteAndCheckExecutionStrategy.cs]: I think `SimpleResult` is not clear as well, "raw" implies that no changes or modifications are applied to the result, it is literally the result from the execution of the code. Maybe `CodeExecutionResult` or something in that direction should be considered..

**Pull request merged**